### ISSUE TITLE
Add a --skip-dev-dependencies flag

### ISSUE THREAD

**Issue by JohnBerd** [depcheck/depcheck#713]
### Feature request description

There are too many false positives in devDependencies React Native project, and I would like to add this tool to my CI without having to change the .depcheckrc each time I add a dev dependency.

It would be great to have an option to skip the devDependencies

Cheers!

**Comment by rumpl** [comment]
So we actually used to have this option, then we deprecated it and finally removed it, there were too many issues with it... You can see here: https://github.com/depcheck/depcheck/pull/450

What false positives do you get?

**Comment by JohnBerd** [comment]
Thank you for your quick answer!

Such a shame. Sorry for not having found this issue, so I have exactly the same problem.
For example with babel, types, typescript, eslint modules.
Anyway I don't want the devDependencies to be checked at all.

**Comment by ma-zal** [comment]
Dear code owners,
I am also looking for `--skip-dev-dependencies` flag implementation, but for another case:

I would like to use it to identify the incorrect situation, where some "dev dependency" is used in production code. Example of my expected usage: `depcheck . --ignore-patterns test* --skip-dev-dependencies`.

*I am able to make code changes and provide the Pull Request for this new feature. But first, I would like to get "approval" from code owners that this feature makes sense for him.*

**Comment by harunurhan** [comment]
If `devDependencies` is used as intended, there is not much value validating those in the code. It would be nice to have the option at least.

**Comment by dragoscirjan** [comment]
Here's a funny example for this. Please note the fact that depcheck sees itself as a unused dependency.

Context is a bit more complex, however, I do see the need for the `--skip-dev-deps` flag.

```json
  "devDependencies": {
    "@jscpd/html-reporter": "^4.0.1",
    "@templ-project/core": "^0.0.0",
    "@types/license-checker": "^25.0.6",
    "@types/lodash.merge": "^4.6.9",
    "@types/ora": "^3.2.0",
    "@types/stream-buffers": "^3.0.7",
    "depcheck": "^1.4.7",
    "dependency-cruiser": "^16.3.2",
    "jscpd": "^4.0.4",
    "license-checker": "^25.0.1"
  }
```
 
 ```bash
 $ ../../node_modules/.bin/depcheck .
Unused devDependencies
* @jscpd/html-reporter
* @types/jest
* depcheck
* dependency-cruiser
* jscpd
* license-checker
```
